lib/livebook/zta/basic_auth.ex

```diff
@@ -0,0 +1,35 @@
+defmodule Livebook.ZTA.BasicAuth do
+  use GenServer
+  require Logger
+
+  defstruct [:identity_key]
+
+  def start_link(options) do
+    name = Keyword.fetch!(options, :name)
+    GenServer.start_link(__MODULE__, options, name: name)
```

Comment:
I think you can skip the GenServer in this case.

```suggestion
  def start_link(options) do
    name = Keyword.fetch!(options, :name)
    identity_key = Keyword.fetch!(options, :identity_key)
    :persistent_term.put({__MODULE__, name}, identity_key)
    :ignore
```

And then read from persistent term. Please give it a try, I am not 100% sure. :D

Reply:
If you remove the GenServer, you will also need to remove `use GenServer` and add this function instead:

```
  def child_spec(opts) do
    %{id: __MODULE__, start: {__MODULE__, :start_link, [opts]}}
  end
```

I just pushed similar changes to Tailscale. :)

Reply:
You can check the Tailscale implementation as one example.